phoenix-core/src/main/java/org/apache/phoenix/compile/QueryCompiler.java

```diff
@@ -559,8 +559,16 @@ protected QueryPlan compileSingleFlatQuery(StatementContext context, SelectState
         groupBy = groupBy.compile(context, innerPlanTupleProjector);
         context.setResolver(resolver); // recover resolver
         RowProjector projector = ProjectionCompiler.compile(context, select, groupBy, asSubquery ? Collections.<PDatum>emptyList() : targetColumns, where);
-        OrderBy orderBy = OrderByCompiler.compile(context, select, groupBy, limit, offset, projector,
-                groupBy == GroupBy.EMPTY_GROUP_BY ? innerPlanTupleProjector : null, isInRowKeyOrder);
+        OrderBy orderBy = OrderByCompiler.compile(
+                context,
+                select,
+                groupBy,
+                limit,
+                offset,
+                projector,
+                groupBy == GroupBy.EMPTY_GROUP_BY ? innerPlanTupleProjector : null,
+                groupBy == GroupBy.EMPTY_GROUP_BY ? isInRowKeyOrder : true,
+                where);
```

Comment:
Would be ideal if the change could be isolated to OrderByCompiler being aware of its inner plan to determine correctly whether the OrderBy can be compiled out. Any ideas, @maryannxue?

Reply:
Yes, it is a nice suggestion, I would make a modification

Reply:
Eliminating order-by based on the inner plan ordering is definitely the right and ultimate solution, and we should get rid of all “dirty fix” flags here. The compile result, in our case, the QueryPlan should have a definite ordering whether it’s from an order by, or a group-by, or a natural row order, or an order-by from the inner query. However, letting QueryPlan contain that information might not be a good idea. So we should see if we can infer the ordering from a QueryPlan.

Reply:
@maryannxue , yes, you are right , Eliminating order-by based on the inner plan ordering is ultimate solution.  Openning this JIRA is just to optimize the OrderBy for ClientAggregatePlan when there is a GroupBy, it is only to need  to conside the GroupBy of the outer query, and innerQuery is not required to take into account. It is much simpler than purely optimizing OrderBy for ClientScanPlan, which need to  make a huge modification and refactor with OrderByCompiler and OrderPreservingTracker. I have planned to  purely optimize OrderBy for ClientScanPlan after this JIRA by openning a new JIRA.